internal/app/recordtester/recordtester_app.go

```diff
@@ -341,6 +302,111 @@ func (rt *recordTester) Start(fileName string, testDuration, pauseDuration time.
 	// <-downloader.Done()
 	// glog.Infof(`Pulling stopped after %s`, time.Since(started))
 	// exit(55, fileName, *fileArg, err)
+
+	glog.Info("Record testing done, waiting for Asset to be available")
+	if err = rt.isCancelled(); err != nil {
+		return 0, err
+	}
+
+	assetId := ""
+	wait_interval := 30 * time.Second
+	elapsed_time := 0 * time.Second
+asset:
+	for {
+		glog.Infof("Waiting %s for asset, elapsed=%s", wait_interval, elapsed_time)
+		time.Sleep(wait_interval)
+		elapsed_time = elapsed_time + wait_interval
+		if err = rt.isCancelled(); err != nil {
+			return 0, err
+		}
+
+		assets, err := rt.lapi.ListAssets() // assets are retrieved in order of most recent first
+		if err != nil {
+			glog.Errorf("Error listing assets err=%v", err)
+			return 241, err
+			// exit(241, fileName, *fileArg, err)
+		}
+		for _, asset := range *assets { // TODO: only need to check first n assets due to ordering
+			if asset.Name == fmt.Sprintf("live-to-vod-%s", sess.ID) {
+				assetId = asset.ID
+				break asset
+			}
+		}
+
+		// TODO: timeout required here?
+	}
+
+	task_name := fmt.Sprintf("live-to-vod-transcode-%s", sess.ID)
+	createTranscodeTaskResp, _ := rt.lapi.CreateTranscodeTask(assetId, task_name, livepeerAPI.StandardProfiles[0])
+	if err != nil {
+		glog.Errorf("Error creating transcode task err=%v", err)
+		return 242, err
+		// exit(242, fileName, *fileArg, err)
+	}
+	transcodeTask := createTranscodeTaskResp.Task
+	transcodeAsset := createTranscodeTaskResp.Asset
+	glog.Infof("Asset is available id=%s, transcoding taskId=%s outputAssetId=%s", assetId, transcodeTask.ID, transcodeAsset.ID)
+
+	wait_interval = 15 * time.Second
+	elapsed_time = 0 * time.Second
+	for {
+		glog.Infof("Waiting %s for transcode output asset id=%s, elapsed=%s", wait_interval, transcodeAsset.ID, elapsed_time)
+		time.Sleep(wait_interval)
+		elapsed_time = elapsed_time + wait_interval
+		if err = rt.isCancelled(); err != nil {
+			return 0, err
+		}
+
+		asset, err := rt.lapi.GetAsset(transcodeAsset.ID)
+		if err != nil {
+			glog.Errorf("Error retrieving asset id=%s err=%v", transcodeAsset.ID, err)
+			return 243, err
+			// exit(243, fileName, *fileArg, err)
+		}
+		if asset.Status == "ready" {
+			break
+		}
+		if asset.Status != "waiting" {
+			glog.Errorf("Error transcoding asset id=%s, task id=%s outputAssetId=%s err=%v", assetId, transcodeTask.ID, transcodeAsset.ID, err)
+			return 244, err
+			// exit(244, fileName, *fileArg, err)
+		}
+
+		// TODO: timeout required here?
+	}
+
+	createExportTaskResp, _ := rt.lapi.CreateExportTask(transcodeAsset.ID)
+	if err != nil {
+		glog.Errorf("Error creating export task err=%v", err)
+		return 245, err
+		// exit(245, fileName, *fileArg, err)
+	}
+	exportTask := createExportTaskResp.Task
+	glog.Infof("Transcode output asset id=%s ready, exporting, taskId=%s", transcodeAsset.ID, exportTask.ID)
+
+	wait_interval = 5 * time.Second
+	elapsed_time = 0 * time.Second
+	for {
```

Comment:
For these wait intervals, you should also consider using a `time.Ticker` instead for go-ishness. e.g.
```suggestion
	ticker := time.NewTicker(5 * time.Second)
	for range ticker {
	   ...
```